internal/features/stacks/decoder/path_reader.go

```diff
@@ -35,28 +48,46 @@ func (pr *PathReader) PathContext(path lang.Path) (*decoder.PathContext, error)
 
 	switch path.LanguageID {
 	case ilsp.Stacks.String():
-		return stackPathContext(record)
+		return stackPathContext(record, pr.StateReader, pr.ModuleReader)
 	case ilsp.Deploy.String():
 		return deployPathContext(record)
 	}
 
 	return nil, fmt.Errorf("unknown language ID: %q", path.LanguageID)
 }
 
-func stackPathContext(record *state.StackRecord) (*decoder.PathContext, error) {
+func stackPathContext(record *state.StackRecord, stateReader stackschema.StateReader, moduleReader stackschema.ModuleReader) (*decoder.PathContext, error) {
 	// TODO: this should only work for terraform 1.8 and above
 	version := record.RequiredTerraformVersion
 	if version == nil {
-		version = stackschema.LatestAvailableVersion
+		version = tfschema.LatestAvailableVersion
 	}
 
 	schema, err := stackschema.CoreStackSchemaForVersion(version)
 	if err != nil {
 		return nil, err
 	}
 
+	sm := stackschema.NewStackSchemaMerger(schema)
+	sm.SetStateReader(stateReader)
+	sm.SetModuleReader(moduleReader)
```

Comment:
In line with the [comment on terraform-schema](https://github.com/hashicorp/terraform-schema/pull/376#discussion_r1695506664), what do you think about having only one reader in the schema merger?